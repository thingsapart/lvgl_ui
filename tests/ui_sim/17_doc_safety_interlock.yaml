# TICKS: 4
- type: data-binding
  state:
    - safety_key_on: false
    - x_pos: 100.0
    - jog_speed: 10.0
  actions:
    - key_turned:
        toggle: safety_key_on
    - jog_x_plus:
        when:
          condition: [and, safety_key_on, [>, x_pos, 0]]
          then:
            - dec: { x_pos: jog_speed }
  schedule:
    - { tick: 1, action: jog_x_plus }
    - { tick: 2, action: key_turned }
    - { tick: 3, action: jog_x_plus }
    - { tick: 4, action: jog_x_plus }
