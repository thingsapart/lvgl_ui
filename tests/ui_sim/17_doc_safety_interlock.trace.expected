--- UI-Sim Trace Start ---
STATE_SET: jog_speed = 10.000 (old: null)
STATE_SET: safety_key_on = false (old: null)
STATE_SET: x_pos = 100.000 (old: null)
NOTIFY: jog_speed = 10.000
NOTIFY: safety_key_on = false
NOTIFY: x_pos = 100.000

--- TICK 1 ---
ACTION: jog_x_plus value=null

--- TICK 2 ---
ACTION: key_turned value=null
STATE_SET: safety_key_on = true (old: false)
NOTIFY: safety_key_on = true

--- TICK 3 ---
ACTION: jog_x_plus value=null
STATE_SET: x_pos = 90.000 (old: 100.000)
NOTIFY: x_pos = 90.000

--- TICK 4 ---
ACTION: jog_x_plus value=null
STATE_SET: x_pos = 80.000 (old: 90.000)
NOTIFY: x_pos = 80.000

--- UI-Sim Trace End ---
